libs/muddle/internal/muddle_fake.hpp

```diff
@@ -0,0 +1,564 @@
+#pragma once
+//------------------------------------------------------------------------------
+//
+//   Copyright 2018-2019 Fetch.AI Limited
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+//
+//------------------------------------------------------------------------------
+
+#include "muddle.hpp"
+#include "muddle/muddle_endpoint.hpp"
+
+#include "muddle_logging_name.hpp"
+
+namespace fetch {
+namespace muddle {
+
+/**
+ * Construct a global in-memory network which can be used by fake muddles.
+ *
+ * To do this, create a map of address to a collection of packets. Peers can then 'send'
+ * packets by writing into this collection. If the receiving muddle is listening, it has
+ * a responsibility to collect the packets.
+ *
+ */
+class FakeNetwork
```

Comment:
For testing, a 'fake muddle' + network is created. Will allow network-esque tests to be run in parallel without port limitations.